lib/plain/endpoints/webhook.ts

```diff
@@ -0,0 +1,101 @@
+import { AxiosInstance } from 'axios'
+import * as raw from './raw'
+import { QueryParams, GetSpaceParams, CollectionProp } from './common-types'
+import { normalizeSelect } from './utils'
+import {
+  WebhookProps,
+  CreateWebhooksProps,
+  WebhookHealthProps,
+  WebhookCallDetailsProps,
+  WebhookCallOverviewProps,
+  UpdateWebhookProps,
+} from '../../entities/webhook'
+import { SetOptional } from 'type-fest'
+import { cloneDeep, clone } from 'lodash'
+
+type GetWebhookParams = GetSpaceParams & { webhookDefinitionId: string }
+
+type CreateWebHookParams = SetOptional<GetWebhookParams, 'webhookDefinitionId'>
+
+type GetWebhookCallDetailsUrl = GetWebhookParams & { callId: string }
+
+const getBaseUrl = (params: GetSpaceParams) => `/spaces/${params.spaceId}/webhook_definitions`
+
+const getWebhookCallBaseUrl = (params: GetSpaceParams) => `/spaces/${params.spaceId}/webhooks`
+
+const getWebhookUrl = (params: GetWebhookParams) =>
+  `${getBaseUrl(params)}/${params.webhookDefinitionId}`
+
+const getWebhookCallUrl = (params: GetWebhookParams) =>
+  `${getWebhookCallBaseUrl(params)}/${params.webhookDefinitionId}/calls`
+
+const getWebhookCallDetailsUrl = (params: GetWebhookCallDetailsUrl) =>
+  `${getWebhookCallBaseUrl(params)}/${params.webhookDefinitionId}/calls/${params.callId}`
+
+const getWebhookHealthUrl = (params: GetWebhookParams) =>
+  `${getWebhookCallBaseUrl(params)}/${params.webhookDefinitionId}/health`
+
+export const get = (http: AxiosInstance, params: GetWebhookParams) => {
+  return raw.get<WebhookProps>(http, getWebhookUrl(params))
+}
+
+export const getManyCallDetails = (http: AxiosInstance, params: GetWebhookParams & QueryParams) => {
+  return raw.get<CollectionProp<WebhookCallOverviewProps>>(http, getWebhookCallUrl(params), {
+    params: normalizeSelect(params.query),
+  })
+}
+
+export const getCallDetails = (http: AxiosInstance, params: GetWebhookCallDetailsUrl) => {
+  return raw.get<WebhookCallDetailsProps>(http, getWebhookCallDetailsUrl(params))
+}
+
+export const getHealthStatus = (http: AxiosInstance, params: GetWebhookParams) => {
+  return raw.get<WebhookHealthProps>(http, getWebhookHealthUrl(params))
+}
+
+export const getMany = (http: AxiosInstance, params: GetSpaceParams & QueryParams) => {
+  return raw.get<CollectionProp<WebhookProps>>(http, getBaseUrl(params), {
+    params: normalizeSelect(params.query),
+  })
+}
+
+export const create = (
+  http: AxiosInstance,
+  params: CreateWebHookParams,
+  rawData: CreateWebhooksProps
+) => {
+  const { webhookDefinitionId } = params
+
+  return webhookDefinitionId
+    ? raw.put<WebhookProps>(http, getWebhookUrl(params as GetWebhookParams), rawData)
+    : raw.post<WebhookProps>(http, getBaseUrl(params), rawData)
+}
+
+export const update = async (
+  http: AxiosInstance,
+  params: GetWebhookParams,
+  rawData: UpdateWebhookProps,
+  headers?: Record<string, unknown>
+) => {
+  const webhookToUpdate = await raw.get<WebhookProps>(http, getWebhookUrl(params))
```

Comment:
What is the rationale behind that kind of implementation? It's deviating from the non-plain implementation and its not really in line with one of the main drivers for creating the plain version of the CMA which is reducing network requests.

Reply:
I think I saw that from a different perspective and that is for sure up for a discussion. If the cycle of an entity is fetch -> change -> update then it does not make sense to have this extra network call. But it seems like we have always want that cycle of fetch the entity -> change the data -> update the data.